server/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -1040,6 +1059,34 @@ public MetaData build() {
             }
 
             // build all indices map
+
+            SortedMap<String, AliasOrIndex> aliasAndIndexLookup = buildAliasAndIndexLookup();
+            AliasValidator.validateAliasWriteOnly(aliasAndIndexLookup);
+
+            // set any `is_write_index == null` to their appropriate boolean values
+            for (AliasOrIndex aliasOrIndex : aliasAndIndexLookup.values()) {
+                if (aliasOrIndex.isAlias()) {
+                    AliasOrIndex.Alias alias = (AliasOrIndex.Alias) aliasOrIndex;
+                    updateIndicesWithDefaultWriteIndex(alias);
+                }
+            }
+
+            aliasAndIndexLookup = Collections.unmodifiableSortedMap(buildAliasAndIndexLookup());
+
+
+            // build all concrete indices arrays:
+            // TODO: I think we can remove these arrays. it isn't worth the effort, for operations on all indices.
```

Comment:
I believe this is about memory allocation - I think we can discuss this but in a different issue.

Reply:
I didn't write this